tldraw/tldraw
# feat(snaps): add multiple snap lines and viewport filtering

This PR:

- Adds multiple snap lines, if more than one is found
- Filters snappable shapes based on the viewport

![Kapture 2021-10-21 at 19 14 07](https://user-images.githubusercontent.com/23072548/138336209-1edbc2ba-ac7b-4432-9bb6-65dd6afce2bc.gif)

### Change type


### Test plan

1. Drag shapes to trigger snapping and verify multiple snap lines appear when aligned with multiple objects.
2. Verify snapping performance/behavior when many shapes are off-screen.


### Release notes

- Improved snapping with support for multiple snap lines and better performance via viewport filtering.
